client/packages/invoices/src/Returns/DetailView/DetailView.tsx

```diff
@@ -32,34 +32,39 @@ export const DetailView: FC = () => {
   const { data, isLoading } = useReturns.document.outboundReturn();
   const t = useTranslation('distribution');
   const navigate = useNavigate();
-  // const onRowClick = useCallback(
+
+  const onRowClick = () => {};
+
+  // useCallback(
   //   (item: StockOutLineFragment | StockOutItem) => {
   //     onOpen({ item: toItemRow(item) });
   //   },
-  //   [toItemRow, onOpen]
+  //   [toItemRow]
   // );
-  // const onAddItem = (draft?: Draft) => {
+  const onAddItem = () => {};
+  //  (draft?: Draft) => {
   //   onOpen(draft);
   //   setMode(ModalMode.Create);
   // };
 
   if (isLoading) return <DetailViewSkeleton hasGroupBy={true} hasHold={true} />;
 
-  // const tabs = [
-  //   {
-  //     Component: (
-  //       <ContentArea
-  //         // onRowClick={!isDisabled ? onRowClick : null}
-  //         onAddItem={onAddItem}
-  //       />
-  //     ),
-  //     value: 'Details',
-  //   },
-  //   {
-  //     Component: <ActivityLogList recordId={data?.id ?? ''} />,
-  //     value: 'Log',
-  //   },
-  // ];
+  const tabs = [
+    {
+      Component: (
+        <ContentArea
+          onRowClick={onRowClick}
+          onAddItem={onAddItem}
+          rows={data?.lines?.nodes}
```

Comment:
I don't really understand why Typescript is complaining about `lines` not existing -- you can clearly see it exists on the type when hovering over `data`:
![Screenshot 2024-02-29 at 12 00 02 PM](https://github.com/msupply-foundation/open-msupply/assets/5456533/90da22f4-912b-468a-aa65-3db3ab1f6ff8)

Reply:
Ah this is due to the query actually returning either an `InvoiceNode` or a `NodeError` (it's a bit frustrating becuase VSCode hides the `NodeError` part behind a `...` 😅) 

If you take a look in the `packages/invoices/src/Returns/api/api.ts`, you'll see some of the other methods have a type check step `if (__typename === 'InvoiceNode') return ...`, otherwise throwing an error - if you chuck one of these in the `outboundReturnByNumber` method Typescript should be satisfied!

Reply:
Ah yes, I thought it would be something to do with the union type, didn't know where to put the check. Thanks 👍